fixed role and non-existing-kind

VedantMahabaleshwarkar opened the pull request:
This is a PR related to the issue mentioned here : #13

barney-s commented on config/manager/kustomization.yaml:
```diff
@@ -1,2 +1,7 @@
+kind: Kustomization
+images:
+- name: controller
+  newName: quay.io/vedantm/airflow-operator
```
@kaxil  - This would benefit from a recommended docker registry for hosting airflow related docker images.

VedantMahabaleshwarkar replied:
Ah yes this shouldn't be in the PR, my bad.

barney-s commented on controllers/airflowcluster_controller.go:
```diff
@@ -66,16 +66,16 @@ type AirflowClusterReconciler struct {
 // +kubebuilder:rbac:groups=storage.k8s.io,resources=storageclasses,verbs=get;list;watch;create;update;patch;delete
 
-// +kubebuilder:rbac:groups=airflow.apache.org,resources=airflowclusters,verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups=airflow.apache.org,resources=airflowclusters;airflowclusters/finalizers,verbs=get;list;watch;create;update;patch;delete
```
what is this finalizers subresource ? Is that a standard one ?

vpavlin replied:
It is standard for OpenShift, so that it is able to add ownerReferences to the owned resources. It will do no harm when deployed on Kubernetes

barney-s commented on config/rbac/role.yaml:
```diff
@@ -6,7 +6,9 @@ metadata:
 rules:
-- resources:
+- apiGroups:
+  - ""
```
Why is this needed ?

VedantMahabaleshwarkar replied:
This fixes things for Openshift clusters. Kubernetes allows an empty apiGroups field but Openshift does not. Figured that adding it wouldn't hurt because everything will work on Kubernetes just the same even after the change.

vpavlin replied:
Yup, basically validation fails on OpenShift because it uses a bit tighter model

barney-s commented on Dockerfile:
```diff
@@ -21,6 +22,7 @@ RUN CGO_ENABLED=0 GOOS=linux GOARCH=amd64 GO111MODULE=on go build -a -o manager
 # Refer to https://github.com/GoogleContainerTools/distroless for more details
 FROM gcr.io/distroless/static:nonroot
 WORKDIR /
+COPY templates/ templates/
```
Does this suffice ? I saw permission denied

vpavlin replied:
@VedantMahabaleshwarkar can you confirm? 

I believe this is it, it worked for us...where did you see permission denied?

VedantMahabaleshwarkar replied:
I didn't see permission denied anywhere. Before the change the Custom Resource had the error  `Missing templates/*, no such file or directory` and adding this line fixed the error.

turbaszek commented:
@VedantMahabaleshwarkar could you please rebase onto actual master? We've introduced some static code checks:
https://github.com/apache/airflow-on-k8s-operator/blob/master/docs/development.md#static-checks

VedantMahabaleshwarkar commented:
> @VedantMahabaleshwarkar could you please rebase onto actual master? We've introduced some static code checks:
> https://github.com/apache/airflow-on-k8s-operator/blob/master/docs/development.md#static-checks

@turbaszek I have rebased my PR
